tests/test_views.py

```diff
@@ -0,0 +1,927 @@
+        assert response.templates[0].name == 'nomination/metadata_report.html'
+
+    @pytest.mark.parametrize('metadata_with_valueset', [
+        True,
+        False
+    ])
+    def test_context(self, client, metadata_with_valueset):
+        if metadata_with_valueset:
+            project = factories.ProjectWithMetadataFactory(metadata1=None)
+        else:
+            project = factories.ProjectWithMetadataFactory(metadata2=None)
+        metadata = models.Project_Metadata.objects.first().metadata
+        met_value = models.Value.objects.first()
+        value = met_value.key if metadata_with_valueset else u'something'
+        factories.URLFactory.create_batch(
+            3,
+            url_project=project,
+            attribute=metadata.name,
+            value=value
+        )
+        response = client.get(reverse('field_report', args=[project.project_slug, metadata.name]))
+        namelist = [(met_value.value if metadata_with_valueset else value, 3, value)]
+
+        assert response.context['project'] == project
+        assert str(response.context['valdic']) == str([(value, 3)])
+        assert response.context['namelist'] == namelist
+        assert response.context['field'] == metadata.name
+
+
+class TestValueReport():
+
+    def test_status_and_content_type(self, rf):
+        project = factories.ProjectFactory()
+        request = rf.get('/')
+        response = views.value_report(request, project.project_slug, '', '')
+
+        assert response.status_code == 200
+        assert response['Content-Type'] == 'text/plain;'
+
+    @pytest.mark.parametrize('val, url_value', [
+        ('asdf', 'asdf'),
+        ('asdf', 'asdf/')
+    ])
+    def test_report_text(self, rf, val, url_value):
+        project = factories.ProjectFactory()
+        field = 'qwerty'
+        urls = factories.URLFactory.create_batch(
+            3,
+            url_project=project,
+            attribute=field,
+            value=url_value
+        )
+        request = rf.get('/')
+        response = views.value_report(request, project.project_slug, field, val)
+
+        assert '#This list of URLs' in response.content
+        for url in urls:
+            assert url.entity in response.content
+
+
+class TestNominatorReport():
+
+    @pytest.mark.parametrize('field', [
+        'nominator',
+        'institution'
+    ])
+    def test_status_ok(self, rf, field):
+        project = factories.ProjectFactory()
+        request = rf.get('/')
+        response = views.nominator_report(request, project.project_slug, field)
+
+        assert response.status_code == 200
+
+    @pytest.mark.parametrize('field', [
+        'nominator',
+        'institution'
+    ])
+    def test_template_used(self, client, field):
+        project = factories.ProjectFactory()
+        response = client.get(reverse('nominator_report', args=[project.project_slug, field]))
+        assert response.templates[0].name == 'nomination/nominator_report.html'
+
```

Comment:
The rationale was that since those ('nominator' and 'institution') are explicitly defined in the urls.py as the two options for the URL, that perhaps I should make sure they both lead to the same template. However, I'm not firm on that stance.